rm-datasource/src/main/java/io/seata/rm/datasource/AbstractPreparedStatementProxy.java

```diff
@@ -103,7 +101,10 @@ public AbstractPreparedStatementProxy(AbstractConnectionProxy connectionProxy, P
      * @param x     the x
      */
     protected void setParamByIndex(int index, Object x) {
-        parameters[--index].add(x);
+        if (!parameters.containsKey(--index)) {
```

Comment:
What's the purpose of op `--`? It looks inconsistent with method `getParamsByIndex `.

Reply:
The base class method of set()  will pass param from 1,and we should ensure that it is taken from 0, so we handle the index with --.

Reply:
We should keep the index parameter of `setParamByIndex` and `getParamByIndex` the same meaning.